### PR TITLE
[model] feat: support dit trainer & wan e2e training

### PR DESCRIPTION
**Pull request by Coach257**
# Main feature
1. e2e training wan2.1B on tom-and-jerry dataset
2. support online_training/offline_embedding/offline_training

Read docs [‎docs/examples/wan2.1_I2V_1.3B.md](https://github.com/ByteDance-Seed/VeOmni/pull/570/changes#diff-cd87a7b1f4b43a56248f95bc4f722d417abf28fd35dd2676fb8fa6c1845ca050) and [docs/usage/support_new_models/dit_model_guide.md](https://github.com/ByteDance-Seed/VeOmni/pull/570/changes#diff-ce73b2b760a0e9e45e982b6693f0b29733721c93e39c547a461dae71facf663f) for more details.

Related to #533.

### CODE REVIEW

File: veomni/trainer/callbacks/checkpoint_callback.py [hunk 1]
```diff
@@ -168,3 +179,41 @@ def _save_checkpoint(self, state: TrainerState):
...
+            output_dir=args.train.checkpoint.output_dir,
+            ckpt_manager=args.train.checkpoint.manager,
+        )
+
+        model_state_dict = get_peft_model_state_dict(self.trainer.model, model_state_dict)
+        # save bf16 lora
+        model_state_dict = {k: v.to(torch.bfloat16) for k, v in model_state_dict.items()}
+        lora_adapter_save_path = os.path.join(save_checkpoint_path, "adapter_model.bin")
+        os.makedirs(save_checkpoint_path, exist_ok=True)
+        _save_state_dict(model_state_dict, lora_adapter_save_path, safe_serialization=False)
```
[the remark this answers is not in the document]
Comment (Coach257) [reply veomni/trainer/callbacks/checkpoint_callback.py:214]: peft load adapter_model.bin
